Null SSLSocket causes NPE when jdbc url contains "ssl=true"

<!-- delete unnecessary items -->
### Describe the bug
The DummySSLConnectionSocketFactory creates a null SSLSocket because initSslContext in HttpAPIClientHelper#createHttpClient is false. 
The root cause is that JdbcConfiguration does not read the ssl property from jdbc url.

### Steps to reproduce
1. set dependency version between 0.8.0 to 0.8.2
2. set jdbc url like: jdbc:clickhouse://localhost:8123/testdb?ssl=true

### Expected behaviour
SSLSocket not null

### Code example
```java
url: jdbc:clickhouse://localhost:8123/testdb?ssl=true
username: xxx
password: xxx
driver-class-name: com.clickhouse.jdbc.ClickHouseDriver

<dependency>
    <groupId>com.clickhouse</groupId>
    <artifactId>clickhouse-jdbc</artifactId>
    <version>0.8.0</version>
</dependency>
```
### Error log
Caused by: java.lang.NullPointerException: SSL Socket
java.base/java.util.Objects.requireNonNull(Objects.java:259)
 org.apache.hc.core5.util.Args.notNull(Args.java:165)
 org.apache.hc.core5.http.impl.io.SocketHolder.<init>(SocketHolder.java:58)
 org.apache.hc.client5.http.impl.io.DefaultManagedHttpClientConnection.bind(DefaultManagedHttpClientConnection.java:188)
 org.apache.hc.client5.http.impl.io.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:232)
 org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:490)
 org.apache.hc.client5.http.impl.classic.InternalExecRuntime.connectEndpoint(InternalExecRuntime.java:164)
 org.apache.hc.client5.http.impl.classic.InternalExecRuntime.connectEndpoint(InternalExecRuntime.java:174)
 org.apache.hc.client5.http.impl.classic.ConnectExec.execute(ConnectExec.java:144)
 org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51)
 org.apache.hc.client5.http.impl.classic.ProtocolExec.execute(ProtocolExec.java:192)
 org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51)
 org.apache.hc.client5.http.impl.classic.ContentCompressionExec.execute(ContentCompressionExec.java:150)
 org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51)
 org.apache.hc.client5.http.impl.classic.HttpRequestRetryExec.execute(HttpRequestRetryExec.java:113)
 org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51)
 org.apache.hc.client5.http.impl.classic.RedirectExec.execute(RedirectExec.java:110)
 org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51)
 org.apache.hc.client5.http.impl.classic.InternalHttpClient.doExecute(InternalHttpClient.java:174)
 org.apache.hc.client5.http.impl.classic.CloseableHttpClient.execute(CloseableHttpClient.java:87)
 org.apache.hc.client5.http.impl.classic.CloseableHttpClient.execute(CloseableHttpClient.java:55)
 org.apache.hc.client5.http.classic.HttpClient.executeOpen(HttpClient.java:183)
 com.clickhouse.client.api.internal.HttpAPIClientHelper.executeRequest(HttpAPIClientHelper.java:423)

### Configuration
#### Environment
* Client version: 0.8.0 - 0.8.2
* Language version: jdk21
* OS: win11

--- a/jdbc-v2/src/main/java/com/clickhouse/jdbc/internal/JdbcConfiguration.java
+++ b/jdbc-v2/src/main/java/com/clickhouse/jdbc/internal/JdbcConfiguration.java
@@ -61,7 +61,9 @@ public class JdbcConfiguration {
         initProperties(urlProperties, info);
 
         // after initializing all properties - set final connection URL
-        boolean useSSL = Boolean.parseBoolean(info.getProperty(DriverProperties.SECURE_CONNECTION.getKey(), "false"));
+        boolean useSSLInfo = Boolean.parseBoolean(info.getProperty(DriverProperties.SECURE_CONNECTION.getKey(), "false"));
+        boolean useSSLUrlProperties = Boolean.parseBoolean(urlProperties.getOrDefault(DriverProperties.SECURE_CONNECTION.getKey(), "false"));
+        boolean useSSL = useSSLInfo || useSSLUrlProperties;
         String bearerToken = info.getProperty(ClientConfigProperties.BEARERTOKEN_AUTH.getKey(), null);
         if (bearerToken != null) {
             clientProperties.put(ClientConfigProperties.BEARERTOKEN_AUTH.getKey(), bearerToken);

--- a/jdbc-v2/src/test/java/com/clickhouse/jdbc/internal/JdbcConfigurationTest.java
+++ b/jdbc-v2/src/test/java/com/clickhouse/jdbc/internal/JdbcConfigurationTest.java
@@ -18,6 +18,7 @@ import static org.testng.Assert.assertEquals;
 
 public class JdbcConfigurationTest {
 
+
     @Test(dataProvider = "testConnectionUrlDataProvider")
     public void testConnectionUrl(String jdbcUrl, String connectionUrl, Properties properties, Map<String, String> expectedClientProps) throws Exception {
         JdbcConfiguration configuration = new JdbcConfiguration(jdbcUrl, properties);
@@ -38,7 +39,7 @@ public class JdbcConfigurationTest {
         Map<String, String> useSSLParams = Map.of("ssl", "true");
         Map<String, String> withListParams = Map.of("database", "default", "param1", "value1", "custom_header1", "val1,val2,val3", "user", "default", "password", "");
         Map<String, String> withListParamsQuotes = Map.of("database", "default", "param1", "value1", "custom_header1", "\"role 1,3,4\",'val2',val3", "user", "default", "password", "");
-
+        Map<String, String> useDatabaseSSLParams = Map.of("database", "clickhouse", "ssl", "true", "user", "default", "password", "");
 
         return new Object[][] {
                 {"jdbc:clickhouse://localhost:8123/", "http://localhost:8123", defaultProps, defaultParams},
@@ -47,7 +48,7 @@ public class JdbcConfigurationTest {
                 {"jdbc:clickhouse://localhost:8443/", "https://localhost:8443", useSSL, useSSLParams},
                 {"jdbc:clickhouse://localhost:8443/default?param1=value1&custom_header1=val1,val2,val3", "http://localhost:8443", defaultProps, withListParams},
                 {"jdbc:clickhouse://localhost:8443/default?custom_header1=\"role 1,3,4\",'val2',val3&param1=value1", "http://localhost:8443", defaultProps, withListParamsQuotes},
-
+                {"jdbc:clickhouse://localhost:8443/clickhouse?ssl=true", "https://localhost:8443", defaultProps, useDatabaseSSLParams},
         };
     }
 

